lib/import/brca/providers/st_george/st_george_handler.rb

```diff
@@ -184,137 +184,131 @@ def process_r208(record)
+            elsif gene.blank? && gene_other.present?
+              process_other_gene(genes, genotype, genotypes, record)
+            else
+              process_status_genes(genotype, 4, @all_genes, genotypes)
+            end
+          end
 
-            gene_list.each do |gene_value|
-              mapped_gene_values = BRCA_GENE_MAP[gene_value] || []
+          def process_genes(genes, genotype, genotypes)
+            pos_gene = genes['gene']
+            process_status_genes(genotype, 2, pos_gene, genotypes)
+            remaining_genes = @all_genes - pos_gene
+            process_status_genes(genotype, 1, remaining_genes, genotypes)
+          end
 
-              mapped_gene_values.each do |value|
-                if value == gene
-                  status = /#{gene_value}\s?\(?fail\)?/i.match?(record.raw_fields['gene (other)']) ? 9 : 1
-                  genotype.add_status(status)
-                else
-                  genotype.add_status(1)
+          def process_genes_with_other(gene_other, genes, genotype, genotypes)
+            pos_gene = genes['gene']
+            process_status_genes(genotype, 2, pos_gene, genotypes)
+            remaining_genes = @all_genes - pos_gene
 
-                end
-              end
+            if gene_other.match(/Fail/i)
+              failed_gene = genes['gene (other)']
+              process_status_genes(genotype, 9, failed_gene, genotypes)
+              remaining_genes -= failed_gene
             end
 
-            true
+            process_status_genes(genotype, 1, remaining_genes, genotypes)
           end
 
-          def update_status(status1, status2, column, column_name, genotype)
-            # update genotype status depending on if the gene is in the same column that the rule applies to
+          def process_other_gene(genes, genotype, genotypes, record)
+            pos_gene = @gene_other_brca_size == 1 ? genes['gene (other)'] : genes['variant dna']
+            if pos_gene.present?
+              process_status_genes(genotype, 2, pos_gene, genotypes)
+              remaining_genes = @all_genes - pos_gene
+              process_status_genes(genotype, 1, remaining_genes, genotypes)
+            else
+              assign_status_on_gene_other(record, genes, genotype, genotypes)
+            end
+          end
 
-            if column == column_name
-              genotype.add_status(status1)
+          def assign_status_on_gene_other(record, genes, genotype, genotypes)
+            binding.pry if record.raw_fields['servicereportidentifier'] == 'V001777'
+            gene_other = record.raw_fields['gene (other)']
+            remaining_genes = @all_genes
+            case gene_other
+            when /Fail/i
+              process_failed_gene_other(genes, genotype, genotypes, remaining_genes)
+            when /\?\z/i
+              process_status_genes(genotype, 4, genes['gene'], genotypes)
+            when /^c\.|^Ex.*Del\z|^Ex.*Dup\z|^Het\sDel|^Het\sDup/ix
```

Comment:
As confirmed by Fiona, regex needs to account for e.g. 'Hetdel' entries as well, it's currently assigning this as test status 4 but needs to be 2.

Reply:
Thanks @NImeson , Updated and test added in [add51ac](https://github.com/NHSDigital/data_management_system/pull/77/commits/add51ac1fd9d6771e9538f3d2a9490eb2e3e314b)